mantidproject/mantid
# Project Recovery: Sanitise the # charecter in workspace names

**Description of work.**
Stops the project recovery from using # in any of it's scripts and also sorts an issue when created workspaces do not match original name list.

**To test:**

* Run the following script:
```
CreateWorkspace(DataX=range(12), DataY=range(12), DataE=range(12), NSpec=4, OutputWorkspace='0Reb#inned')
for i in range(100):
    RenameWorkspace(InputWorkspace='%sReb#inned'%str(i), OutputWorkspace='%sReb#inned'%str(i+1))
for i in range(300):
    CloneWorkspace(InputWorkspace='100Reb#inned', OutputWorkspace='%sCl#o#ne'%str(i))
```
*  Allow project recovery to start and finish
* Crash Mantid using the segfault algorithm
* Restart Mantid; select `Yes` when presented with recovery options
* Check that the workspaces are re-loaded, e.g. see if `232Cl#o#ne` exists as a workspace

Fixes #23248  


#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
